msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -324,65 +359,45 @@ private void initializeApplication() {
-
-    /**
-     * Returns the PublicClientConfiguration for this instance of PublicClientApplication
-     * Configuration is based on the defaults established for MSAl and can be overridden by creating the
-     * PublicClientApplication using {@link PublicClientApplication#PublicClientApplication(Context, int)}
-     *
-     * @return
-     */
-    public PublicClientApplicationConfiguration getConfiguration() {
-        return mPublicClientConfiguration;
-    }
-
-    /**
-     * Listener callback for asynchronous loading of msal IAccount accounts.
+     * Listener callback for asynchronous initialization of IPublicClientApplication object.
      */
-    public interface AccountsLoadedCallback {
-
+    public interface ApplicationCreatedListener {
         /**
-         * Called once Accounts have been loaded from the cache.
-         *
-         * @param accounts The accounts in the cache.
+         * Called once an IPublicClientApplication is successfully created.
          */
-        void onAccountsLoaded(List<IAccount> accounts);
+        void onCreated(final IPublicClientApplication application);
     }
 
     /**
-     * Listener callback for asynchronous loading of msal IAccount accounts.
+     * Listener callback for asynchronous loading of MSAL mode retrieval.
      */
-    public interface AccountsRemovedCallback {
-
+    public interface BrokerAccountModeCallback {
         /**
-         * Called once Accounts have been removed from the cache.
-         *
-         * @param isSuccess true if the account is successfully removed.
+         * Called once MSAL mode is retrieved from Broker.
+         * If the value can't be retrieved, this will fall back to the BROKER_ACCOUNT_MODE_MULTIPLE_ACCOUNT mode.
          */
-        void onAccountsRemoved(Boolean isSuccess);
+        void onGetMode(String mode);
     }
 
     /**
-     * Listener callback for asynchronous loading of broker AccountRecord accounts.
+     * @return The current version for the sdk.
      */
-    public interface BrokerAccountsLoadedCallback {
-        /**
-         * Called once Accounts have been loaded from the broker.
-         * @param accountRecords The accountRecords in broker.
-         */
-        void onAccountsLoaded(List<AccountRecord> accountRecords);
+    public static String getSdkVersion() {
+        return BuildConfig.VERSION_NAME;
     }
 
     /**
-     * Asynchronously returns a List of {@link IAccount} objects for which this application has RefreshTokens.
+     * Returns the PublicClientConfiguration for this instance of PublicClientApplication
+     * Configuration is based on the defaults established for MSAl and can be overridden by creating the
+     * PublicClientApplication using {@link PublicClientApplication#create(Context, int, ApplicationCreatedListener)}
      *
-     * @param callback The callback to notify once this action has finished.
+     * @return
      */
+    public PublicClientApplicationConfiguration getConfiguration() {
```

Comment:
Should this move up to the interface?

Reply:
I'm slightly inclined to the 'no' side. The reason is that I want the interface to only contains API that are meant to be used by app developers. Not strong opinion tho.

Reply:
I was thrown off by the `public` access modifier here - can we remove move this down to default? I think this is only called by tests which should be in the same pkg

Reply:
will do.